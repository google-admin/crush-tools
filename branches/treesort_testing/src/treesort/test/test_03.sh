test_number=03
description="all column sort"

subtest_desc=("AA,BB,CC,DD"
							"BB,AA,CC,DD"
							"DD,AA,CC,BB"
							"3,2,4,1 indexes"
							'AA,^BB,CC,^DD'
							'AA,^BB,^CC,DD'
							'^3,1,^4,2 indexes'
							'^3,^4,^2,^1 indexes'
							)
subtest_opts=(
'-K AA,BB,CC,DD'
'-K BB,AA,CC,DD'
'-K DD,AA,CC,BB'
'-k 3,2,4,1 -p'
'-K AA,^BB,CC,^DD'
'-K AA,^BB,^CC,DD'
'-k ^3,1,^4,2 -p'
'-k ^3,^4,^2,^1 -p'
)

for subtest in `seq 0 7`; do
  expected="$test_dir/test_$test_number.$subtest.expected"
  outfile="$test_dir/test_$test_number.$subtest.actual"

  $bin ${subtest_opts[$subtest]} \
       "$test_dir/test2.in" \
       > "$outfile"

  if [ $? -ne 0 ] ||
     [ "`diff -q $outfile $expected`" ]; then
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" FAIL
  else
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" PASS
    rm "$outfile"
  fi
done
