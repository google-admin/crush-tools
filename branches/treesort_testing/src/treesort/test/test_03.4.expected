AAþBBþCCþDD
119þs5FX4ERZP8þ2009-10-19þ20.0623
119þqHbD7IgvBsþ2009-10-15þ19.8086
119þoj-npio3Cqþ2009-10-18þ11.9550
119þkX69CHRCpFþ2009-10-29þ16.2961
119þczHP0Y09gOþ2009-10-18þ25.4415
119þVjx-O1uUiyþ2009-10-06þ23.1071
119þQ99ceOGe58þ2009-10-15þ14.6088
119þPKndexdB5Yþ2009-10-06þ43.1227
119þNqFghFb16Pþ2009-10-19þ25.4415
119þMLySuT_6L-þ2009-10-25þ21.4784
119þD6DavOvkp5þ2009-10-21þ13.3454
119þ-_WgFpelwTþ2009-10-15þ21.4784
120þcOA-GflWf9þ2009-10-02þ19.8086
120þbNfKeDJp35þ2009-10-15þ13.3454
120þ_okpFO60P5þ2009-10-19þ21.4784
120þZs9uH9wqwvþ2009-10-19þ21.4784
120þY0_LegXodZþ2009-10-18þ14.6088
120þXdQde3c2LRþ2009-10-29þ16.2961
120þWxgda_Rz1uþ2009-10-23þ19.8086
120þS-BUVAIarsþ2009-10-02þ23.1071
120þQOhQYF2L4Uþ2009-10-18þ20.0623
120þOcgRMuIUVWþ2009-10-09þ25.4415
120þN1CuDuqLsqþ2009-10-25þ19.8086
120þ8ZDolJWFhKþ2009-10-19þ25.4415
120þ5-dmH-yGymþ2009-10-29þ11.9550
120þ1Qbw9NbEK2þ2009-10-19þ19.8086
222þqdigsUU1EHþ2009-10-21þ11.9550
222þl0sIDKi1Mnþ2009-10-09þ43.1227
222þdCzebYQfiHþ2009-10-18þ13.3454
222þ_Dh-EwmwHUþ2009-10-29þ13.3454
222þZeP1N6ZHo8þ2009-10-09þ11.9550
222þZ_RNo9lPThþ2009-10-09þ21.4784
222þR0lHfvqeE2þ2009-10-18þ20.0623
222þQddzqzZhvaþ2009-10-19þ20.0623
222þOxNCXf_QO5þ2009-10-15þ11.9550
222þNrNOPtKMh2þ2009-10-09þ23.1071
222þMk8GiTvT-2þ2009-10-21þ19.8086
222þHF90x6_QiBþ2009-10-23þ13.3454
222þGkijsWrtDmþ2009-10-25þ16.2961
222þ8HPzHGZLDfþ2009-10-02þ43.1227
379þzHaJcq1brbþ2009-10-21þ43.1227
379þtH_urSAL6Iþ2009-10-15þ20.0623
379þpUbcEzSLg5þ2009-10-15þ43.1227
379þmdg21XYvE4þ2009-10-18þ13.3454
379þi4Vn-mjUX_þ2009-10-25þ20.0623
379þccfVWBl1Ifþ2009-10-15þ25.4415
379þbWuILAzu5Lþ2009-10-06þ23.1071
379þUH3YCtEsnEþ2009-10-21þ20.0623
379þDx5ojQn-G6þ2009-10-19þ13.3454
379þ1aRiJ5zvaUþ2009-10-15þ14.6088
379þ-l43yW3QQXþ2009-10-19þ21.4784
428þw-DsqhmkP4þ2009-10-06þ16.2961
428þi0-lOQWHXeþ2009-10-18þ23.1071
428þQg_963dGw2þ2009-10-23þ25.4415
428þQEIlvFBOAEþ2009-10-21þ14.6088
428þJw-SmuJJwHþ2009-10-29þ16.2961
428þDMqu6KjLMnþ2009-10-09þ21.4784
428þ0J2H9eMxz5þ2009-10-19þ19.8086
525þpfetgY4yYtþ2009-10-29þ21.4784
525þ_mNRE6q64Dþ2009-10-29þ23.1071
525þV_Srbg1Xgkþ2009-10-09þ19.8086
525þMP3BenV_PWþ2009-10-02þ19.8086
525þH3laCVXVqFþ2009-10-29þ20.0623
525þFJFwWbUCEgþ2009-10-18þ16.2961
525þAnKSv96_ykþ2009-10-29þ16.2961
525þ-ynhLNQjXnþ2009-10-18þ43.1227
525þ-ca4VazCCoþ2009-10-19þ43.1227
692þuGBjnSxQwIþ2009-10-19þ23.1071
692þooYbIsNctcþ2009-10-21þ14.6088
692þkC3oWoj1_8þ2009-10-25þ21.4784
692þj-OdwoZTzmþ2009-10-21þ23.1071
692þgplXXJQCbDþ2009-10-23þ20.0623
692þQOa1zDDXrcþ2009-10-19þ21.4784
692þK26asHqM1Bþ2009-10-25þ21.4784
692þBuPJhoNfZYþ2009-10-21þ13.3454
692þ3Abgi5FvpKþ2009-10-18þ16.2961
704þlAkjpi7B0uþ2009-10-21þ11.9550
704þgbwjbbmOR5þ2009-10-25þ13.3454
704þab_ALFxdBlþ2009-10-23þ16.2961
704þ_6GXpFrx7Mþ2009-10-09þ21.4784
704þV-dWH8XhzNþ2009-10-15þ11.9550
704þPb0jUNXPGFþ2009-10-21þ11.9550
704þCwEfrzVzlNþ2009-10-29þ19.8086
790þyj8FAR6sT8þ2009-10-19þ21.4784
790þpPrA9-ibvPþ2009-10-19þ23.1071
790þn1pETdd_e2þ2009-10-25þ25.4415
790þghqsjQH7CAþ2009-10-25þ21.4784
790þeGWkG9dnHHþ2009-10-29þ13.3454
790þcEXl50cdsGþ2009-10-02þ23.1071
790þU_DsBKeZBOþ2009-10-15þ16.2961
790þSEiwhaVJcvþ2009-10-25þ20.0623
790þQHJtym0emOþ2009-10-19þ14.6088
790þ0CjQNzoprAþ2009-10-21þ25.4415
876þvT9F16VWxuþ2009-10-02þ16.2961
876þqv6ulolvVmþ2009-10-18þ14.6088
876þnLKFOKtBt5þ2009-10-19þ21.4784
876þZ971TGdirGþ2009-10-21þ23.1071
876þRHEIKKC-eNþ2009-10-06þ16.2961
876þOypnZVbJOYþ2009-10-15þ13.3454
876þ6RIZ_xP77Oþ2009-10-23þ21.4784
